packages/driver/src/cy/commands/request.coffee

```diff
@@ -125,6 +125,15 @@ module.exports = (Commands, Cypress, cy, state, config) ->
       if originOrBase = config("baseUrl") or cy.getRemoteLocation("origin")
         options.url = $Location.qualifyWithBaseUrl(originOrBase, options.url)
 
+      ## Make sure the url unicode characters are properly escaped
+      ## https://github.com/cypress-io/cypress/issues/5274
+      try
+        options.url = decodeURI(options.url)
+      catch URIError
+        # The url is not already encoded just silence the error
+      finally
+        options.url = encodeURI(options.url)
```

Comment:
Hmm, this also seems to encode the domain of URLs passed in:

```
>> encodeURI('http://example-漢字.com/?a=漢字')
"http://example-%E6%BC%A2%E5%AD%97.com/?a=%E6%BC%A2%E5%AD%97"
```

I'm not even sure if non-punycoded Unicode domains work in Cypress today, but either way, I don't think that the domain should be urlencoded. Just the `path` component of the URL should be urlencoded, right?

Reply:
You totally right.

Thinking about it, I'm wondering if doing an:
`options.url = new URL()`

Which will percent escape the path and convert the domain Unicode to `xn--example--d44...` into the `.href` attribute would not be a better way to handle this.

Reply:
I've changed the way I handle the URL, by using the `new URL()`  constructor instead of `encodeURI` allow to both, properly encode unicode in domain (with `xn...`) and percent escape unicode in pathname.